include/uv/errno.h

```diff
@@ -456,8 +456,14 @@
 # define UV__ESOCKTNOSUPPORT (-4025)
 #endif
 
+/* FreeBSD defines ENODATA in /usr/include/c++/v1/errno.h which is only visible
+ * if C++ is being used. Define it directly to avoid problems when integrating
+ * libuv in a C++ project.
```

Comment:
"C project"? A C++ project would get the correct definition (9919) whereas a C project gets our fallback value, right?

Reply:
That's correct but when building nodejs I observed that libuv was built with C whereas Nodejs with C++ which caused that libuv code (namely `uv_err_name_r()`) was seeing `UV_ENODATA` as -4024 whereas accessing that same value from C++ saw -9919. At least that was how I interpreted that behavior.

Reply:
Fair enough.